org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/archetype/MavenArchetype.java

```diff
@@ -11,16 +11,21 @@
  * Contributors:
  *      Christoph Läubrich - initial API and implementation
  *******************************************************************************/
-package org.eclipse.m2e.core.ui.internal.archetype;
+package org.eclipse.m2e.core.ui.archetype;
 
 import org.apache.maven.archetype.catalog.Archetype;
 
 import org.eclipse.m2e.core.project.IArchetype;
 
 
-public class MavenArchetype implements IArchetype {
+/**
+ * Implementation of {@link IArchetype} based on {@link Archetype}
+ * 
+ * @since 2.1.0 (package version 1.0.0)
+ */
+public final class MavenArchetype implements IArchetype {
```

Comment:
Why is this required as API? I don't see it used in the signatures of ArchetypeManager/ArchetypeGenerator

Reply:
IArchetype is used as method parameter, so someone would need to come up with an implementation.

Reply:
See my comment, beside that a custom implementation is really trivial...

Reply:
Done in https://github.com/eclipse-m2e/m2e-core/pull/922/commits/c39ab6b3ee2ade42b2a7da710f82006791c1c38b

Reply:
> Done in [c39ab6b](https://github.com/eclipse-m2e/m2e-core/commit/c39ab6b3ee2ade42b2a7da710f82006791c1c38b)

This looks good to me!